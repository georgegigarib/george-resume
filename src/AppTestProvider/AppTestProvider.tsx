import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
