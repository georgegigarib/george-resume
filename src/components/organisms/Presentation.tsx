import React from 'react'
import { useTranslation } from 'react-i18next'

export default function Presentation(): React.ReactElement {
  const { t } = useTranslation()

  return (
    <div className="flex relative flex-col justify-between h-full p-4  rounded-xl bg-gray-200 dark:bg-gray-800">
      <div className="mb-4">
        <h2 className="text-xl font-bold mb-2 text-gray-700 dark:text-gray-300">
          {t('me.presentation.title')}
        </h2>
        <p className="text-gray-700 dark:text-gray-300">{t('me.presentation.description')}</p>
      </div>
    </div>
  )
}
