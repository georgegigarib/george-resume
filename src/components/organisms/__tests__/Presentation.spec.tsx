import { render } from '@testing-library/react'
import { describe, expect, it } from 'vitest'
import Presentation from '@/components/organisms/Presentation'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'

describe('Presentation component', () => {
  const renderComponent = () =>
    render(
      <AppTestProvider>
        <Presentation />
      </AppTestProvider>
    )

  it('matches the snapshot', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})
