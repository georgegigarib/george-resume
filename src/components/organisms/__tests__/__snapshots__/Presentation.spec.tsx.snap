// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Presentation component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex relative flex-col justify-between h-full p-4 rounded-xl bg-gray-200 dark:bg-gray-800"
  >
    <div
      class="mb-4"
    >
      <h2
        class="text-xl font-bold mb-6 text-gray-700 dark:text-gray-300"
      >
        Hi! I'm George Gil
      </h2>
      <p
        class="text-gray-700 dark:text-gray-300"
      >
        I'm a curious software engineer who also plays guitar, not always at the same time though.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
